### PR TITLE
Changes to Ansible role dataset_loader to add ATS 9 support

### PR DESCRIPTION
**Pull request by alexander-cabinian**
This PR includes several modifications to the main.yml file for the Ansible `dataset_loader` role. Work was undertaken to add ATS 9 support to the `dataset_loader` role to allow a version of ATS 9 to be utilized in a lab/testing environment. Previously, ATS 7 and 8 only were supported by the `dataset_loader` role. ATS 9 specific parameters were added where necessary to replace deprecated ATS 7/8 parameters. Other parameters in `main.yml` that caused spurious log messages in ATS 9 were moved to sections for ATS 7 and 8 in the file.

<hr/>

## Which Traffic Control components are affected by this PR?

- Automation - Ansible Roles - `dataset_loader` 

## What is the best way to verify this PR?

1. Determine version of ATS 9 to use in lab environment
2. Utilizing Ansible playbooks, build ATC lab environment
3. On lab caches, confirm ATS 9 is installed, running with no errors
4. On lab caches, confirm common and ATS 9-specific parameters are set in appropriate configuration files

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 1]
```diff
@@ -1297,6 +1325,39 @@ dl_ds_default_profile_layers:
...
+          configFile: records.config
+          value: INT 0
+        # Param value changed between ATS 7,9
+        - name: CONFIG proxy.config.net.sock_option_flag_in
+          configFile: records.config
+          value: INT 1
+        # Param value changed between ATS 8 & 9, default is 0
+        - name: CONFIG proxy.config.ssl.handshake_timeout_in
+          configFile: records.config
+          value: INT 30
```
Comment (jhg03a) [inline infrastructure/ansible/roles/dataset_loader/defaults/main.yml:1360]: Is this to say that the default changed between 8 & 9 and we're preserving the older value of 30 in ATS 9?

**Review by jhg03a — CHANGES_REQUESTED** [review]
Looks good overall, just needs some comment cleanups and one duplicate to figure out

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 2]
```diff
@@ -805,6 +799,14 @@ dl_ds_default_profile_layers:
...
           configFile: plugin.config
           value: ''
+        # ATS 7 specific param
+        - name: CONFIG proxy.config.http.origin_max_connections
+          configFile: records.config
+          value: INT 500
+        # ATS 7 specific param
+        - name: CONFIG proxy.config.cache.http.compatibility.4-2-0-fixup
+          configFile: records.config
+          value: INT 0
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: Modified comment as requested.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 3]
```diff
@@ -813,6 +815,19 @@ dl_ds_default_profile_layers:
...
           configFile: records.config
           value: INT 0
+        # ATS 8 specific param
+        - name: CONFIG proxy.config.http.origin_max_connections
+          configFile: records.config
+          value: INT 500
+        # ATS 8 specific param
+        - name: CONFIG proxy.config.cache.http.compatibility.4-2-0-fixup
+          configFile: records.config
+          value: INT 0
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml:825]: Modified comment as requested.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 4]
```diff
@@ -858,7 +870,20 @@ dl_ds_default_profile_layers:
...
-      v8: []
+        # ATS 7 specific param
+        - name: CONFIG proxy.config.http.origin_max_connections
+          configFile: records.config
+          value: INT 1500
+      v8:
+        # ATS 8 specific param
+        - name: CONFIG proxy.config.http.origin_max_connections
+          configFile: records.config
+          value: INT 1500
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: Modified comment as requested.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 5]
```diff
@@ -1216,6 +1232,18 @@ dl_ds_default_profile_layers:
...
+          configFile: records.config
+          value: INT 100000
+        # ATS 7 specific param
+        - name: CONFIG proxy.config.http.origin_max_connections_queue
+          configFile: records.config
+          value: INT 0
+        # ATS 7 specific param
+        - name: CONFIG proxy.config.http.parent_proxy_routing_enable
+          configFile: records.config
+          value: INT 1
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: Modified comment as requested.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 6]
```diff
@@ -1297,6 +1325,39 @@ dl_ds_default_profile_layers:
         - name: LOCAL proxy.local.cluster.type
           configFile: records.config
           value: INT 3
+        # Param value changed between ATS 7 & 8
+        - name: CONFIG proxy.config.net.sock_option_flag_in
+          configFile: records.config
+          value: INT 5
+        # ATS 8 specific param
+        - name: CONFIG proxy.config.net.max_connections_active_in
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml:1329]: Modified comment as requested.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 7]
```diff
@@ -1297,6 +1325,39 @@ dl_ds_default_profile_layers:
         - name: LOCAL proxy.local.cluster.type
           configFile: records.config
           value: INT 3
+        # Param value changed between ATS 7 & 8
+        - name: CONFIG proxy.config.net.sock_option_flag_in
+          configFile: records.config
+          value: INT 5
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: This parameter was removed in favor of the value given in common/common. No problems observed during lab env buildout with the common/common value.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml [hunk 8]
```diff
@@ -1297,6 +1325,39 @@ dl_ds_default_profile_layers:
...
+          configFile: records.config
+          value: INT 100000
+        # Param change for ATS 9, from proxy.config.http.origin_max_connections_queue
+        - name: CONFIG proxy.config.http.per_server.connection.queue_size
+          configFile: records.config
+          value: INT 0
+        # Param value changed between ATS 7,9
+        - name: CONFIG proxy.config.net.sock_option_flag_in
+          configFile: records.config
+          value: INT 1
```
[the remark this answers is not in the document]
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: This parameter was removed in favor of the value given in common/common. No problems observed during lab env buildout with the common/common value.

File: infrastructure/ansible/roles/dataset_loader/defaults/main.yml (hunk 1 above)
Comment (alexander-cabinian) [reply infrastructure/ansible/roles/dataset_loader/defaults/main.yml]: This parameter was removed from the ATS 9-specfiic section. 

`proxy.config.ssl.handshake_timeout_in` has a default value of 0 in ATS 7 and 8, and a default value of 30 in ATS 9. So, no sense in keeping it in `dataset_loader`.

**Comment by alexander-cabinian** [comment]
@jhg03a , responded to all comments and made appropriate changes where necessary. Please review and provide anything additional you find.

**Review by jhg03a — APPROVED** [review]

**Pull request merged**